crypto/crmf/crmf_lib.c

```diff
@@ -673,12 +675,18 @@ X509 *OSSL_CRMF_ENCRYPTEDVALUE_get1_encCert(OSSL_CRMF_ENCRYPTEDVALUE *ecert,
                 CRMF_R_UNSUPPORTED_CIPHER);
         return NULL;
     }
-
+    /* select symmetric cipher based on algorithm given in message */
+    if ((cipher = EVP_get_cipherbynid(symmAlg)) == NULL) {
+        CRMFerr(CRMF_F_OSSL_CRMF_ENCRYPTEDVALUE_GET1_ENCCERT,
+                CRMF_R_UNSUPPORTED_CIPHER);
+        goto end;
+    }
+    ERR_clear_error();
+    cikeysize = EVP_CIPHER_key_length(cipher);
     /* first the symmetric key needs to be decrypted */
     pkctx = EVP_PKEY_CTX_new(pkey, NULL);
     if (pkctx != NULL && EVP_PKEY_decrypt_init(pkctx)) {
         ASN1_BIT_STRING *encKey = ecert->encSymmKey;
-        size_t eksize = 0;
 
         if (EVP_PKEY_decrypt(pkctx, NULL, &eksize, encKey->data, encKey->length)
                 <= 0
```

Comment:
Almost okay.  I suggest:
```C
                || EVP_PKEY_decrypt(pkctx, ek, &eksize, encKey->data,
                                    encKey->length) <= 0
                || eksize != cikeysize) {
            ERR_clear_error(); /* error state has sensitive information */
            CRMFerr(CRMF_F_OSSL_CRMF_ENCRYPTEDVALUE_GET1_ENCCERT,
                    CRMF_R_ERROR_DECRYPTING_SYMMETRIC_KEY);
            goto end;
}
```
Bonus points, if you can do the `result <= 0 || eksize != cikeysize` in constant time.

Reply:
Hope I am closer to the bonus now. Thanks for your suggestions.